content/post/2020-07-24-data-science-tutorials-with-learnr-and-gradethis.md

```diff
@@ -0,0 +1,73 @@
+With the gradethis package, it's also easy to give helpful feedback if a student gets a foreseen wrong answer, so we can nudge them in the right direction.
+
+This package is not yet on CRAN, but you can install it from the Github repository [here](https://github.com/rstudio-education/gradethis).
+
+
+### Challenges
+
+We quickly learned that converting the homework assignments wouldn't be as straightforward as just replicating the questions in a learnr tutorial. Since some tasks like interpreting visualisations can't easily be graded automatically, we had to make certain trade-offs. Usually, this either meant replacing an open-ended question with a series of multiple-choice questions while trying to preserve a questions essential lesson. 
+
+Another question we had to ask ourselves frequently was whether the student's code itself or the output it produced should be evaluated. Here's an example of this conundrum: 
+
+![Learnr tutorial code block example](/post/2020-07-24-data-science-tutorials-with-learnr-and-gradethis_files/sVa8ehy.png)
+
+There are many ways to answer this question in R, and since we wanted students to be awarded points whichever correct way they chose, we implemented result grading on this question, so that every answer that produced the correct output would be marked as correct. 
+With visualisations and code that doesn't produce any output, however, this becomes a bit trickier. 
+In those situations, we decided to grade the code itself, which has the disadvantage that the code has to be exactly the same as in the solution, even though it would have the same effect if two lines were interchanged. We tried to circumvent the problem by giving the students more hints and providing a structure where fewer mistakes would be possible. The following image is an example of how we dealt with visualisation exercises:
+
+![Learnr tutorial code block showing visualisation exercise example](/post/2020-07-24-data-science-tutorials-with-learnr-and-gradethis_files/qYIvTx1.png)
+
+
+### Conclusion
+
+We hope that our converted assignments will help scale the course and deliver something very close to the experience we had last fall. 
+Both of us learned a lot through this project and we are very eager to hear what next years students have to say about the course! We would like to thank Dr. Çetinkaya-Rundel for the giving us the opportunity to be part of IDS and her neverending patience for us! 
+
+For students to access the tutorials, we're placing them in the [**dsbox**](https://www.github.com/rstudio-education/dsbox) package. They can then be run directly from the *Tutorials* pane in RStudio, like so:
+
+![Tutorials pane in RStudio](/post/2020-07-24-data-science-tutorials-with-learnr-and-gradethis_files/tutorials-pane.PNG)
```

Comment:
@mine-cetinkaya-rundel I've got a screenshot for the tutorials pane. Unfortunately, neither RStudio cloud nor my local installation seems to want to actually run one of the tutorials themselves, (it would appear my talents are limited in certain areas), so if we want more than just a screenshot of the tutorials pane you might need to be the one to take it. sorry... (I am trying to get it working, promise!)